File: api/models/schemas.py
from __future__ import annotations

from typing import Any
from uuid import UUID

from pydantic import BaseModel, Field

from shared.protocol import Task


class JobCreateRequest(BaseModel):
    prompt: str = Field(min_length=1)
    workspace_url: str | None = None
    config: dict[str, Any] = Field(default_factory=dict)


class JobCreateResponse(BaseModel):
    job_id: str
    status: str


class JobDetailResponse(BaseModel):
    job: dict[str, Any]
    tasks: list[Task] = Field(default_factory=list)


class AgentInfo(BaseModel):
    name: str | None = None
    role: str | None = None
    status: str | None = None
    model: str | None = None
    heartbeat_ts: str | None = None
    current_task: str | None = None
    job_id: str | None = None

File: packages/api/src/schemas.ts
import { z } from "zod";

export const JobCreateBody = z.object({
  prompt: z.string().min(1),
  workspace_url: z.string().optional(),
  config: z.record(z.unknown()).default({}),
});
export type JobCreateBody = z.infer<typeof JobCreateBody>;

export const JobCreateResponse = z.object({
  job_id: z.string(),
  status: z.string(),
});
export type JobCreateResponse = z.infer<typeof JobCreateResponse>;

export const AgentInfo = z.object({
  name: z.string().optional(),
  role: z.string().optional(),
  status: z.string().optional(),
  model: z.string().optional(),
  heartbeat_ts: z.string().optional(),
  current_task: z.string().optional(),
  job_id: z.string().optional(),
});
export type AgentInfo = z.infer<typeof AgentInfo>;
